server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -220,6 +220,13 @@ final class DefaultSearchContext extends SearchContext {
             );
             queryBoost = request.indexBoost();
             this.lowLevelCancellation = lowLevelCancellation;
+            SearchSourceBuilder builder = request.source();
```

Comment:
not necessary. `explainScore` is called already, which tells us to explain things. Thus we should use that as a signal to do so.